File: javascript/001_fibonacci_sequence.js
module.exports = {
  calculate: function(index) {
    if (index === 0 || index === 1) { return index; }

    if (index < 0) {
      return this.calculate(index + 2) - this.calculate(index + 1);
    } else {
      return this.calculate(index - 2) + this.calculate(index - 1);
    }
  }
};

File: csharp/001FibonacciSequence.cs
namespace Kata
{
  public class Fibonacci
  {
    public int Calculate(int index)
    {
      if (index == 0 || index == 1)
      {
        return index;
      }

      if (index < 0)
      {
        return Calculate(index + 2) - Calculate(index + 1);
      }
      else
      {
        return Calculate(index - 2) + Calculate(index - 1);
      }
    }
  }
}
